form.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+    function func(String $text, int $size, &$valid, &$errors){
+        $tmp = $_POST["$text"];
+        if(!$tmp){
+            $errors["$text"] = "Полето е задължително."; 
+        } elseif(strlen($tmp) > $size ){
+            $errors["$text"] = "Полето има максимална дължина $size символа.";
+        } else {
+            $valid[$text] = $tmp;
+        }        
+    }
+
+    $valid = array();
+    $errors = array();
+    
+    if( $_POST ){
+        func("title", 150, $valid, $errors);
+        func("teacher",200, $valid, $errors);
+
+        $description = $_POST['description'];
+        if(!$description){
+            $errors['description'] = "Описанието е задължително."; 
+        } elseif(strlen($description) < 10 ){
+            $errors['description'] = "Описанието има минимална дължина 10 символа.";
+        } else {
+            $valid['description'] = $description;
+        }
+
+        $group = $_POST['group'];
+        if(!$group){
+            $errors['group'] = "Не сте избрали група.";
+        } else {
+            $valid['group'] = $group;
+        }
+
+        $credits = $_POST['credits'];
+        if (!$credits){
+            $errors['credits'] = "Не сре задали кредити.";
+        } elseif($credits < 0 || $credits > 20){
+            $errors['credits'] = "Кредити могат да са от 0 до 20.";
+        } else {
+            $valid['credits'] = $credits;
+        }
+
+        if(count($valid) == 5 ){
```

Comment:
What would happen if we add a new requirement and there is an error but the `$valid` count stays the same? Wouldn't it be less error-prone if we are checking for no errors? And also this way we will also eliminate this magical number `5`.